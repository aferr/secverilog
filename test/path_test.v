module path_test(clk);

   input {L} clk;

   reg [2:0] seq {Domain w} w;
   reg [2:0] seq {Domain x} x;
   reg [2:0] seq {Domain w} y;

   reg {L} a;
   reg {L} b;
   reg {L} c;
   reg {L} d;

   reg [2:0] {Domain non_seq} non_seq;
   

   always @(posedge clk) begin
      w <= 3;
      if (a) begin
         if (!b) begin
            if (c && d) begin
               x <= 1;
            end
            y <= 1;
         end
      end
   end // always @ (posedge clk)

   always @(*) begin
      if (a) begin
         if (!b) begin
            non_seq = 2;
         end
      end
   end
   




endmodule
