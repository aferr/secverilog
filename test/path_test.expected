unsat
unsat
unsat
Start base conditions are satisfiable? (should be sat)
sat
End base conditions are satisfiable?
unsat
unsat
unsat
unsat
unsat
unsat
unsat
unsat
unsat
